samcli/lib/samlib/wrapper.py

```diff
@@ -77,6 +82,39 @@ def run_plugins(self, convert_local_uris=True):
     def template(self):
         return copy.deepcopy(self._sam_template)
 
+    def _patch_language_extensions(self) -> None:
+        """
+        Monkey patch SamResource.valid function to exclude checking DeletePolicy
+        and UpdateReplacePolicy when language extensions are set
+        """
+        template_copy = self.template
+        if self._check_using_langauge_extension(template_copy):
+
+            def patched_func(self):
+                if self.condition:
+                    if not is_str()(self.condition, should_raise=False):
+                        raise InvalidDocumentException(
+                            [InvalidTemplateException("Every Condition member must be a string.")]
+                        )
+                return SamResourceType.has_value(self.type)
+
+            SamResource.valid = patched_func
+
+    @staticmethod
+    def _check_using_langauge_extension(template: Dict) -> bool:
+        """
+        Check if language extensions are set in the template's Transform
+        :param template: template to check
+        :return: True if language extensions are set in the template, False otherwise
+        """
+        transform = template.get("Transform")
+        if transform:
+            if isinstance(transform, str) and transform == "AWS::LanguageExtensions":
```

Comment:
It's ok to leave this in, but is this a "can't happen" case? Doesn't SAM CLI verify that the template contains the Serverless transform?

Reply:
This was a case I was a little unsure about. Since SAM-CLI supports raw CFN resources, there's no requirement for a template to have the serverless transform for it to work with SAM. Going off this logic, this seemed like a valid case.

Reply:
Gotcha, makes sense to keep it then. Thanks!